src/client.ts

```diff
@@ -119,15 +127,22 @@ export class Client {
    * Performs a HTTP request to the endpoint specified with the default
    * options of the client.
    *
-   * @param  {String}   resource Resource to be called by GET.
+   * @param  {String}   resource Resource to be called by GET and POST.
+   * @param  {object}   payload  Resource to be called by POST.
    * @return {Promise<Response>}
    */
-  public async request(resource: string): Promise<Response> {
-    const response = await fetch(resource, {
-      method: 'GET',
+  public async request(resource: string, payload?: object): Promise<Response> {
+    const method: string = typeof payload === 'undefined' ? 'GET' : 'POST';
```

Comment:
I would do this:

```typescript
const method: string = payload ? 'POST' : 'GET';
const headers: GenericObject<string> = Object.assign(
  {}, 
  this.headers, 
  payload ? {'Content-Type': 'application/json'} : {}
);
const body: string = payload ? JSON.stringify(payload) : undefined;
const response = await fetch(resource, {
  mode: 'cors',
  method,
  headers,
  body
});
```

For me, we could use `this.headers` to store user headers (no defaults) and have a method like:

```typescript
private __buildHeaders(headers?: GenericObject<string>): GenericObject<string> {
  return Object.assign(
    {
      'Accept': 'application/json'
    }, 
    this.headers, 
    headers || {}
  );
}
```

So the previous code would be:

```typescript
const method: string = payload ? 'POST' : 'GET';
const headers: GenericObject<string> = payload ? {'Content-Type': 'application/json'} : {};
const body: string = payload ? JSON.stringify(payload) : {};
const response = await fetch(resource, {
  mode: 'cors',
  headers: this.__buildHeaders(headers),
  method,
  body
});
```